atlasdb-feign/build.gradle

```diff
@@ -0,0 +1,64 @@
+apply from: "../gradle/publish-jars.gradle"
+apply plugin: 'org.inferred.processors'
+apply plugin: 'com.github.johnrengelman.shadow'
+
+apply from: '../gradle/shared.gradle'
+
+configurations {
+    explicitShadow
+    compile.extendsFrom(explicitShadow)
+    shadow.extendsFrom(explicitShadow)
+
+    // This is used for cleaning up the POM.
+    toBeShaded
+    compile.extendsFrom(toBeShaded)
```

Comment:
what is the difference between `shadow`, `explicitShadow`, and `toBeShaded` here?